Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp

```diff
@@ -73,6 +77,37 @@ void ProcessLauncher::launchProcess()
 {
     IPC::SocketPair socketPair = IPC::createPlatformConnection(IPC::PlatformConnectionOptions::SetCloexecOnServer);
 
+    GUniquePtr<gchar> processIdentifier(g_strdup_printf("%" PRIu64, m_launchOptions.processIdentifier.toUInt64()));
+    GUniquePtr<gchar> webkitSocket(g_strdup_printf("%d", socketPair.client));
+
+#if USE(LIBWPE)
+    // Currently this feature is behind compile time flag but in the future this flag should be removed
+    if (WPEProcessProvider::singleton().isEnabled()) {
+        unsigned nargs = 3;
+        char** argv = g_newa(char*, nargs);
+        unsigned i = 0;
+        argv[i++] = processIdentifier.get();
+        argv[i++] = webkitSocket.get();
+        argv[i++] = nullptr;
+
+        m_processIdentifier = WPEProcessProvider::singleton().launchProcess(m_launchOptions, argv, socketPair.client);
+
+        if (m_processIdentifier > -1) {
+            // Don't expose the parent socket to potential future children.
+            if (!setCloseOnExec(socketPair.client))
```

Comment:
This needs to be done unconditionally. You don't want to leak the fd only when the process fails to launch. You never want to leak fds.

Reply:
Oh wait, sorry, I see you crash below with `g_error()` if `m_processIdentifier <= -1`. So that's fine then.

But it would be less confusing if you instead write `RELEASE_ASSERT(m_processIdentifier > -1);`, that way you can ditch the extra condition.

Reply:
Well, g_error() prints a nice message when the process dies, so there is some value to using it.

You could do:

```
if (m_processIdentifier <= -1)
    g_error("...");

// Then everything below doesn't have to be indented
if (!setCloseOnExec)
```

This is just a code style suggestion. What you have is fine, I just prefer to check for errors first and then return, instead of checking for errors inside the `else`. Slightly easier to read.